La base de datos ha sido actualizada

AugustoZp opened the pull request:
- Se corrigieron algunos campos con atribuciones erróneas.
- Se agregaron algunos campos faltantes a las tablas.

AugustoZp committed:
La base de datos ha sido actualizada

- Se corrigieron algunos campos con atribuciones erróneas.
- Se agregaron algunos campos faltantes a las tablas.

GilbertoChan commented:
Checando la codificación de la base de datos, no encontré algún error, mas que la duda sobre las imágenes que se están manejando por varchar? pero fuera de eso, todo esta bien

Inge-Luis-Chan-Ku commented:
Se valida  codificación de la base de datos, no se encuentra algún error

DIVSZ reviewed (changes requested):
La base de datos la veo bien en general, si acaso solo sería el mencionar algunas modificaciones para que las tengamos en cuenta. 

En La tabla “categoríes” podríamos agregar un campo "unique" en “name”, ya que se van a manejar por marcas, producto line, o figuras articuladas, etc. y pues no debe haber duplicidad como tener 2 apartados para los funkos o nendoroid… Etc.

En la tabla “users” en "email" podemos agregarle un campo “unique” ya que no puedes tener 2 usuarios con el mismo correo o en su caso que este mismo se cree otra cuenta con el mismo correo.

AugustoZp committed:
Campos modificados

Se han tomado en cuenta una revisión.
- Se agregó el atributo UNIQUE al campo email de la tabla users.
- Se agregó el atributo UNIQUE al campo name de la tabla categories

DIVSZ reviewed (approved):
perfecto

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+
+CREATE TABLE `categories` (
+  `id` int(3) NOT NULL,
+  `seller_id` int(3) NOT NULL,
```
Para qué servirá el campo seller_id en las categorías?

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+CREATE TABLE `orders` (
+  `id` int(3) NOT NULL,
+  `user_id` int(3) NOT NULL,
+  `user_email` varchar(55) NOT NULL,
```
Para qué sirve el user_email dentro de orders, si se puede recuperar con el user_id?

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+  `id` int(3) NOT NULL,
+  `user_id` int(3) NOT NULL,
+  `user_email` varchar(55) NOT NULL,
+  `order_address` varchar(200) NOT NULL,
```
Esto está bien, pero como van a enlazar un usuario a una dirección?

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+  `id` int(3) NOT NULL,
+  `order_id` int(3) NOT NULL,
+  `product_id` int(3) NOT NULL,
+  `product_name` varchar(45) NOT NULL,
```
Para qué quiero el nombre del producto aquí, si lo puedo recuperar de la tabla de productos?

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+  `order_id` int(3) NOT NULL,
+  `product_id` int(3) NOT NULL,
+  `product_name` varchar(45) NOT NULL,
+  `order_address` varchar(200) NOT NULL,
```
Para que quiero la dirección de la orden en el detalle, si adicional ya la tengo en la orden?

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+  `product_name` varchar(45) NOT NULL,
+  `order_address` varchar(200) NOT NULL,
+  `state` varchar(50) NOT NULL,
+  `city` varchar(50) NOT NULL,
```
Qué sentido tiene city y state en el detalle de la orden?

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+  `city` varchar(50) NOT NULL,
+  `image` varchar(45) NOT NULL,
+  `quantity` tinyint(2) NOT NULL DEFAULT 1,
+  `total_amount` int(6) NOT NULL
```
Creo que no debe de ser total_amount, debe de ser price

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+
+CREATE TABLE `products` (
+  `id` int(3) NOT NULL,
+  `seller_id` int(3) NOT NULL,
```
Para que necesito un seller_id aquí?

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+  `category_id` int(3) NOT NULL,
+  `product_name` varchar(45) NOT NULL,
+  `price` int(6) NOT NULL,
+  `quantity` int(5) NOT NULL,
```
Le cambiaría el nombre a stock para ser más descriptivo

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+  `short_desc` varchar(100) NOT NULL,
+  `description` varchar(450) NOT NULL,
+  `image` varchar(45) NOT NULL,
+  `available` tinyint(1) NOT NULL DEFAULT 1
```
Si ya tienes quantity o stock, no necesitas avaiable, ya que, si quantity esta en 0, entonces no esta avalable

abraham1669 commented on ecommerce-unid.sql:
```diff
@@ -0,0 +1,314 @@
+  `username` varchar(10) NOT NULL,
+  `email` varchar(55) NOT NULL,
+  `phone_number` bigint(10) NOT NULL,
+  `role` tinyint(1) NOT NULL DEFAULT 2,
```
Si es la tabla de sellers, entonces el rol debe de ser siempre sellers?

abraham1669 reviewed (commented):
¿Quién aprobó esto?
